From 53d29d35cdbcf493f6a9046458947d3e91f01add Mon Sep 17 00:00:00 2001
From: Rui Miguel Silva <rui.silva@linaro.org>
Date: Wed, 1 Feb 2023 16:11:25 +0000
Subject: [PATCH 36/43] efi_boottime: allow to reset a path after boot

Allow to install multiple protocol interfaces in an
already installed root interface.
This may need to be fix in other way, but for now
looks like the get away fix.

Upstream-Status: Pending
Signed-off-by: Rui Miguel Silva <rui.silva@linaro.org>
---
 lib/efi_loader/efi_boottime.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/lib/efi_loader/efi_boottime.c b/lib/efi_loader/efi_boottime.c
index fea4eb7a342e..90f43ff9a62f 100644
--- a/lib/efi_loader/efi_boottime.c
+++ b/lib/efi_loader/efi_boottime.c
@@ -2669,7 +2669,6 @@ efi_install_multiple_protocol_interfaces_int(efi_handle_t *handle,
 				EFI_PRINT("Path %pD already installed\n",
 					  protocol_interface);
 				ret = EFI_ALREADY_STARTED;
-				break;
 			}
 		}
 		ret = EFI_CALL(efi_install_protocol_interface(handle, protocol,
-- 
2.39.2

